pkg/csi/service/wcpguest/controller.go

```diff
@@ -420,6 +453,128 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
+		// Feature is disabled on the cluster
+		return nil, status.Error(codes.InvalidArgument, "File volume not supported.")
+	}
+	// Build the CnsFileAccessConfig instance name and namespace
+	cnsFileAccessConfigInstance := &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{}
+	cnsFileAccessConfigInstanceName := req.NodeId + "-" + req.VolumeId
+	cnsFileAccessConfigInstanceKey := types.NamespacedName{
+		Namespace: c.supervisorNamespace,
+		Name:      cnsFileAccessConfigInstanceName,
+	}
+	// Check whether the CnsFileAccessConfig instance exist in the supervisor cluster
+	if err := c.cnsOperatorClient.Get(ctx, cnsFileAccessConfigInstanceKey, cnsFileAccessConfigInstance); err != nil {
+		if !apierrors.IsNotFound(err) {
+			// Get() on the CnsFileAccessConfig instance failed with different error
+			msg := fmt.Sprintf("failed to get CnsFileAccessConfig instance: %q. Error: %+v", cnsFileAccessConfigInstance.Name, err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+		// Create the CnsFileAccessConfig instance since it is not found
+		cnsFileAccessConfigInstance = &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      cnsFileAccessConfigInstanceName,
+				Namespace: c.supervisorNamespace},
+			Spec: cnsfileaccessconfigv1alpha1.CnsFileAccessConfigSpec{
+				VMName:  req.NodeId,
+				PvcName: req.VolumeId,
+			},
+		}
+		log.Debugf("Creating CnsFileAccessConfig instance: %+v", cnsFileAccessConfigInstance)
+		if err := c.cnsOperatorClient.Create(ctx, cnsFileAccessConfigInstance); err != nil {
+			msg := fmt.Sprintf("failed to create cnsFileAccessConfig: %v. Error: %v", cnsFileAccessConfigInstance.Name, err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+	}
+	log.Debugf("Found CnsFileAccessConfig: %q", cnsFileAccessConfigInstance.Name)
+	if cnsFileAccessConfigInstance.DeletionTimestamp != nil {
+		// When deletionTimestamp is set, CnsOperator is in the process of removing access for this IP.
+		// When that operation is successful, the instance will be deleted. In a subsequent retry, a new instance will be created.
+		msg := fmt.Sprintf("cnsFileAccessConfigInstance %q is getting deleted. A new instance will be created in the subsequent ControllerPublishVolume request", cnsFileAccessConfigInstance.Name)
+		log.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	// Verify if the CnsFileAccessConfig instance has status with done set to true and error is empty
+	if cnsFileAccessConfigInstance.Status.Done && cnsFileAccessConfigInstance.Status.Error == "" {
+		publishInfo := make(map[string]string)
+		for key, value := range cnsFileAccessConfigInstance.Status.AccessPoints {
+			if key == common.Nfsv4AccessPointKey {
+				publishInfo[common.Nfsv4AccessPoint] = value
+				break
+			}
+		}
+		resp := &csi.ControllerPublishVolumeResponse{
+			PublishContext: publishInfo,
+		}
```

Comment:
Set the disk type as file volume in `publishInfo` before returning response? We're doing it for block volume.